### DISCUSSION TITLE
Ill-conditioned matrix when using `StokesFlow`

### DISCUSSION THREAD

**Discussion by kizzle4shaw** [PMEAL/OpenPNM#2833] [Q&A]
I tried to run Stokes flow simulation on an extracted network from Ketton carbonate but I'm encountering an error below. I need assistance in understanding why is this happening.
<img width="750" alt="IMG_8324" src="https://user-images.githubusercontent.com/32852010/228283498-9e8c197e-7c40-4276-8748-57bf44eb9977.png">

**Comment by ma-sadeghi** [comment]
Hi. Please share a MWE (minimum working example) as it's impossible to debug without.

**Reply by kizzle4shaw** [reply]
Sorry for the late reply our time difference is quite huge. 


Here is the code is used:
```python
import imageio
import numpy as np
import porespy as ps
import openpnm as op
import matplotlib.pyplot as plt
ps.visualization.set_mpl_style()
np.random.seed(10)
im = imageio.volread("C:\\Users\\ERIC\\Desktop\\Ketton\\Ketton.tif")
output = ps.networks.snow2(im, phase_alias = None, boundary_width = 3, accuracy = 'standard', voxel_size = 3.00006e-6, sigma=0.4, r_max = 4, peaks = None)
pn = op.io.network_from_porespy(output.network)
pn['pore.diameter'] = pn['pore.inscribed_diameter']
pn['throat.diameter'] = pn['throat.inscribed_diameter']

pn.add_model(propname='throat.hydraulic_size_factors', model = op.models.geometry.hydraulic_size_factors.pyramids_and_cuboids)
pn.add_model(propname='throat.diffusive_size_factors', model = op.models.geometry.diffusive_size_factors.pyramids_and_cuboids)

h = op.utils.check_network_health(pn)

op.topotools.trim(network=pn, pores=h['disconnected_pores'])
op.topotools.trim(network=pn, pores=h['duplicate_throats'])
op.topotools.trim(network=pn, pores=h['isolated_pores'])

phase = op.phase.Phase(network = pn)
phase['pore.viscosity']=1.0
phase.add_model_collection(op.models.collections.physics.basic)
phase.regenerate_models()

inlet = pn.pores('xmin')
outlet = pn.pores('xmax')

flow = op.algorithms.StokesFlow(network=pn, phase=phase)
flow.set_value_BC(pores=inlet, values=1)
flow.set_value_BC(pores=outlet, values=0)
flow.run()
```

the aim was to calculate permeability. this network has been extracted from ketton carbonate using snow2. 

I also have another problem on top of that, in the hydraulic size factors I tried spheres and cylinders initially but I got an error implying that I have throats bigger than the pores.  so i changed to pyramids and cuboids sorry to ask on this issue too.🙂

**Reply by ma-sadeghi** [reply]
No worries, could you also share the "Ketton.tif"?

**Comment by kizzle4shaw** [comment]
I downloaded the image here https://imperialcollegelondon.box.com/v/iccpsim-ketton2015. it's more than 200mb I don't think it can be uploaded here.
